kubectl-fdb/cmd/profile_analyzer.go

```diff
@@ -0,0 +1,170 @@
+/*
+ * profile_analyzer.go
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"text/template"
+
+	fdbv1beta2 "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta2"
+	"github.com/spf13/cobra"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type profileConfig struct {
+	Namespace   string
+	ClusterName string
+	JobName     string
+	CommandArgs string
+}
+
+func newProfileAnalyzerCmd(streams genericclioptions.IOStreams) *cobra.Command {
+	o := newFDBOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:   "analyze-profile",
+		Short: "Analyze FDB shards to find the busiest team",
+		Long:  "Analyze FDB shards to find the busiest team",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			kubeClient, err := getKubeClient(o)
+			if err != nil {
+				return err
+			}
+
+			clusterName, err := cmd.Flags().GetString("fdb-cluster")
+			if err != nil {
+				return err
+			}
+			startTime, err := cmd.Flags().GetString("start-time")
+			if err != nil {
+				return err
+			}
+			endTime, err := cmd.Flags().GetString("end-time")
+			if err != nil {
+				return err
+			}
+			topRequests, err := cmd.Flags().GetInt("top-requests")
+			if err != nil {
+				return err
+			}
+			templateName, err := cmd.Flags().GetString("template-name")
+			if err != nil {
+				return err
+			}
+
+			namespace, err := getNamespace(*o.configFlags.Namespace)
+			if err != nil {
+				return err
+			}
+
+			return runProfileAnalyzer(kubeClient, namespace, clusterName, startTime, endTime, topRequests, templateName)
+		},
+		Example: `
+# Run the profiler for cluster-1 in the default namespace for the provided state and end time.
```

Comment:
Can we add a note here that this plugin is still experimental. There are still many todos including publishing the profile analyse image before making it a stable sub command.